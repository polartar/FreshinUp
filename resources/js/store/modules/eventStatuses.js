import makeRestStore from 'fresh-bus/store/utils/makeRestStore'

export default ({ items, item }) => {
  return makeRestStore('foodfleet/event-statuses', { items, item })
}
