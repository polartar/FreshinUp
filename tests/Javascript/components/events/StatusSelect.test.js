import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/events/StatusSelect.vue'
import { FIXTURE_EVENT_STATUSES } from 'tests/__data__/eventStatuses'

describe('Event StatusSelect component', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('options set', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          options: FIXTURE_EVENT_STATUSES
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Computed', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('activeItem', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          value: 2,
          options: FIXTURE_EVENT_STATUSES
        }
      })
      expect(wrapper.vm.activeItem.id).toEqual(2)
    })
  })
})
