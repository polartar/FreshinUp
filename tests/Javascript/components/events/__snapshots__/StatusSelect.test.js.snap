// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event StatusSelect component Visuals options set 1`] = `
<div
  class="v-menu"
>
  <button
    class="v-btn theme--dark grey grey-text--text"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      
  Draft
 
      <i
        aria-hidden="true"
        class="v-icon v-icon--right material-icons theme--dark"
      >
        arrow_drop_down
      </i>
    </div>
  </button>
  <div
    class="v-menu__content theme--light "
    style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
  >
     
    <div
      class="v-list theme--light"
      role="list"
    >
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Draft
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            FF Initial Review
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Customer Agreement
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Fleet Member Selection
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Customer Review
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Fleet Member Contracts
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Confirmed
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Cancelled
          </div>
        </a>
      </div>
      <div
        role="listitem"
      >
        <a
          class="v-list__tile v-list__tile--link theme--light"
        >
          <div
            class="v-list__tile__title"
          >
            Past
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
